backend/src/main/java/de/adrianaschepers/shishabuddies/controller/UserController.java

```diff
@@ -61,6 +66,72 @@ public UserController(UserService userService) {
         return ok(createdSettings);
     }
 
+    @PostMapping("create-setup")
+    public ResponseEntity<Setup> createSetup(@RequestBody Setup setup, @AuthenticationPrincipal UserEntity authUser){
+        SetupEntity setupEntity = map(setup);
+        SetupEntity createdSetupEntity = userService.saveSetup(setupEntity,authUser);
+        Setup createdSetup = map(createdSetupEntity);
+        return ok(createdSetup);
+    }
+
+    @GetMapping("all-setups")
+    public ResponseEntity<List<Setup>> getAllSetups(@AuthenticationPrincipal UserEntity authUser){
+        List<SetupEntity> setups = userService.getAllSetups(authUser);
+        if(setups==null){
+            return notFound().build();
+        }
+        return ok(mapSetup(setups));
+    }
+
+    @GetMapping("{title}/setup")
+    public ResponseEntity<Setup> getSetupByTitle(@PathVariable String title,@AuthenticationPrincipal UserEntity authUser){
+        SetupEntity setupEntity = userService.getSetup(authUser,title);
+        Setup searchedSetup = map(setupEntity);
+        return ok(searchedSetup);
+
+    }
+
+    private List<Setup> mapSetup(List<SetupEntity> setupEntities){
+        List<Setup> setups = new LinkedList<>();
+        for (SetupEntity setupEntity: setupEntities) {
+            setups.add(map(setupEntity));
+        }
+        return setups;
+
+    }
+
+    private Setup map(SetupEntity createdSetupEntity) {
```

Comment:
Lass uns da die Tage mal schauen, ob wir dir die ganzen Mapper in nen eigenen Service packen können :)